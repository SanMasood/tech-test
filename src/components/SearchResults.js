import React from 'react';
import '../styles/SearchResults.css';
import PropTypes from 'prop-types';

const SearchResults = ( { image } ) => {

return (
    

    <div className="SearchResults">
        <div className="test">

            <img src={image} data-testid="image-id"
            alt="mockAlt"
            className="card-photo"
            />        
        </div>
    
    </div>

)
    
}
SearchResults.propTypes = {
    image: PropTypes.func.isRequired,
}

export default SearchResults;