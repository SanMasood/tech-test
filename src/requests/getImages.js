import axios from 'axios';

const getImages = (query) => {
    if (!query){
        return Promise.resolve([]);           
            
    }
    else{
        const request = query.toLowerCase(); 

        return axios
        .get(`https://images-api.nasa.gov/search?q=`+request)
        .then((response)=>{
            

            let imageResults = response.data.collection.items;

            let parsedImages = imageResults.filter((images)=> images.data[0].media_type==="image");

            const images = parsedImages.map(image => image.links[0].href);
            //console.log (`Images: ${images}`); 
             return images;   
           

        })
        .catch((err)=>{
            alert("Something went wrong. Please refresh your browser");           
        });        
    }
}
export default getImages;