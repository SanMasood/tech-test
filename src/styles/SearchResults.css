
 
  
  /* Ephxtra-wide grid-posts */
  
/*.search-results{
    border: 2px solid orange;
    margin: 5%;
}
.search-results:hover{
    border: 1px solid blue;
}*/
.SearchResults {
    display: inline-grid;
    grid-column: repeat(3, 25%);
    grid-row: auto;
    margin: 25px;
}

.card-photo{

    border-radius: 25px;
    width: 250px;
    height: 200px;
    object-fit: cover;
    padding: 15px;
   
  }
 
  
 /* 
 .card-photo {
   padding: 0 5px;
    float: left;  
    max-width: 30%;
    max-height: 20%;
    
  }*/
 
  /*.searchResults {
    line-height: 0;
   
  -webkit-column-count: 5;
  -webkit-column-gap:   0px;
  -moz-column-count:    5;
  -moz-column-gap:      0px;
  column-count:         5;
  column-gap:           0px; 
    
  }
  
  .card-photo {    
     width: 20% !important;
  height: 30%;
  }*/
  