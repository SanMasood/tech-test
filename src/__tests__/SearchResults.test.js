import React from 'react';
import axios from 'axios';

import { render, fireEvent, screen } from '@testing-library/react';

import SearchResults from '../components/SearchResults';
import { image, getImages } from '../requests/getImages';

jest.mock('axios');

describe ("SearchResults", () => {

    it("renders correctly", () => {
      const {asFragment} = render (
        <SearchResults        
        />
      )
        expect(asFragment()).toMatchSnapshot();  
      });
      
      /*it("displays all results", async() => {
        const mockResponse = [
          {
            href : "link1",            
          },
          {
            href: "link2",
          },
          {
            href: "link3"
          }
        ]
        const response = await getImages('moon');
        expect(response).toHaveLength(3);
        
      });*/

  
    it("returns thumbnail images of search query", async() => {

      axios.get.mockResolvedValue({
        data: [
          {href: "https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg", rel: "preview", render: "image"},

          {href: "https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg", rel: "preview", render: "image"}

        ]
      })

      const results = await SearchResults({image});
      expect(results).toMatchSnapshot();
    });
    


})