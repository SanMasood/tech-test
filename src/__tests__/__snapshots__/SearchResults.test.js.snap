// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="input-form"
    >
      <label>
        Search:
        <input
          class="search-text"
          type="text"
        />
      </label>
      <button
        class="search-button"
        type="submit"
      >
        Go
      </button>
    </form>
  </div>
</DocumentFragment>
`;
